File: P1/P1/Ejercicio3/MagoBuilder.py
from PersonajeBuilder import PersonajeBuilder
from Armadura import Armadura
from ArmaduraSimple import ArmaduraSimple
from PlantaDecorador import PlantaDecorador

class MagoBuilder(PersonajeBuilder):
    def add_arma(self):
        self.personaje.arma = "Bolas de magía"

    def add_armadura(self):
        self.personaje.armadura = ArmaduraSimple("Armadura Basica")

    def add_habilidad(self):
        self.personaje.habilidad = "Magía"
    
    def set_armadura(self, armadura):
        self.personaje.armadura = armadura
        
    def get_armadura(self):
        return self.personaje.armadura

File: P2/ejercicio3/lib/mago_builder.dart
import 'package:ejercicio3/armadura_simple.dart';
import 'package:ejercicio3/personaje_builder.dart';
import 'package:ejercicio3/armadura.dart';

class MagoBuilder extends PersonajeBuilder{

  @override
  void addArmadura(){
    super.personaje?.armadura = ArmaduraSimple("Armadura Básica");
  }

  @override
  void addHabilidad(){
    super.personaje?.habilidad = "Magía";
  }

  @override
  void addArma(){
    super.personaje?.arma = "Barita mágica";
  }

  void setArmadura(Armadura armadura){
    super.personaje?.armadura = armadura;
  }

  Armadura? getArmadura(){
    return super.personaje?.armadura;
  }
}